packages/Adapters/C_API/test/tstPOD_C_API.cpp

```diff
@@ -85,17 +81,61 @@ TEUCHOS_UNIT_TEST( C_API, create_apply_delete_map )
                  false );
 
   DTK_Map_delete( dtk_map );
+}
+
+//---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( C_API, create_apply_delete_map )
+{
+  // set options using JSON format
+  std::string const options = "{ "\
+    "\"Map Type\": \"Moving Least Square Reconstruction\", "\
+    "\"Basis Type\": \"Wendland\", "\
+    "\"Basis Order\": 2, "\
+    "\"Search Type\": \"Radius\", "\
+    "\"RBF Radius\": 0.3 }";
+
+  // run the test
+  std::vector<double> tgt_field, gold_data;
+  run_random_test( options, tgt_field, gold_data );
 
   // check the target against the gold values
   double const rel_tol = 1e-8;
   double const abs_tol = 1e-6;
-  for (unsigned i = 0; i < field_dim*tgt_num; ++i)
+  unsigned size = gold_data.size();
+  for (unsigned i = 0; i < size; ++i)
+  {
+    TEST_FLOATING_EQUALITY( gold_data[i], tgt_field[i], rel_tol );
+    TEST_COMPARE( std::abs(gold_data[i] - tgt_field[i]), <, abs_tol );
+  }
+}
+
+//---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( C_API, knn_map )
```

Comment:
This is the same as the previous ``TEUCHOS_UNIT_TEST ``, ``options`` is the only thing that changes.
I'd advocate for a ``for (std::string const & options : { ... }) { ... }``